src/slang-printers/types.d.ts

```diff
@@ -0,0 +1,10 @@
+import type { Doc } from 'prettier';
+
+interface PrintSeparatedOptions {
+  firstSeparator?: Doc;
+  separator?: Doc;
+  lastSeparator?: Doc;
```

Comment:
question: is it ever legal to have a "lastSeparator" in Solidity? my understanding is that it is always a parse error.

Reply:
hehe welcome @OmarTawfik.

`lastSeparator` here refers to what happens just after the list is printed:

```Solidity
struct A {
    uint a; // firstSeparator is new line
    uint b;
    uint c; // lastSeparator would be a new line
}

assembly {
    let
        long, // firstSeparator is new line
        list,
        of,
        variable,
        names = f() // last separator would be empty string before the equal.
}
```

In most cases `lastSeparator` matches `firstSeparator` but in some cases it does not and it's necessary the extra parameter.

If you want to propose a better name to not have the `separator` word I'd happily update the PR with that.

Reply:
Thanks for explaining! I misunderstood how it is used. Up to you in this case.